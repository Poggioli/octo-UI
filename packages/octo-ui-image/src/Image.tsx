import { forwardRef } from "@octo-ui/core";
import { StyledImage, StyledRatio } from "./styles";
import { ImageProps } from "./types";

const Image = forwardRef<typeof StyledImage, ImageProps>(
  ({ ratio = 1, ...props }, forwardedRef) => {
    return (
      <StyledRatio ratio={ratio}>
        <StyledImage {...props} ref={forwardedRef} />
      </StyledRatio>
    );
  }
);
Image.toString = () => `.${StyledImage.className}`;

export { StyledImage };
export { Image };
