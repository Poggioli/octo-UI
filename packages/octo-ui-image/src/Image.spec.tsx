import { render } from "@testing-library/react";
import { Image } from "./Image";

describe("<Image />", () => {
  it(`GIVEN a <Image />
      WHEN render is called
      THEN should render a <Image />`, () => {
    const rendered = render(
      <Image
        src="https://images.unsplash.com/photo-1535025183041-0991a977e25b?w=300&dpr=2&q=80"
        alt="Landscape photograph by Tobias Tullius"
        ratio={16 / 9}
      />
    );
    expect(rendered.container).toMatchSnapshot();
  });

  it(`GIVEN a <Image />
      WHEN render is called
      THEN should render a <Image />`, () => {
    const rendered = render(
      <Image
        src="https://images.unsplash.com/photo-1535025183041-0991a977e25b?w=300&dpr=2&q=80"
        alt="Landscape photograph by Tobias Tullius"
      />
    );
    expect(rendered.container).toMatchSnapshot();
  });

  it(`GIVEN a <Image />
      WHEN toString is called
      THEN return a string`, () => {
    expect(Image.toString()).toBe(".octo-UI-c-kxYyuZ");
  });
});
