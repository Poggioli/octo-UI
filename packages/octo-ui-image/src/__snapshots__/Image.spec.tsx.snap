// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Image /> GIVEN a <Image />
      WHEN render is called
      THEN should render a <Image /> 1`] = `
<div>
  <div
    data-radix-aspect-ratio-wrapper=""
    style="position: relative; width: 100%; padding-bottom: 56.25%;"
  >
    <div
      class="octo-UI-c-PJLV"
      style="position: absolute; top: 0px; right: 0px; bottom: 0px; left: 0px;"
    >
      <img
        alt="Landscape photograph by Tobias Tullius"
        class="octo-UI-c-kxYyuZ"
        src="https://images.unsplash.com/photo-1535025183041-0991a977e25b?w=300&dpr=2&q=80"
      />
    </div>
  </div>
</div>
`;

exports[`<Image /> GIVEN a <Image />
      WHEN render is called
      THEN should render a <Image /> 2`] = `
<div>
  <div
    data-radix-aspect-ratio-wrapper=""
    style="position: relative; width: 100%; padding-bottom: 100%;"
  >
    <div
      class="octo-UI-c-PJLV"
      style="position: absolute; top: 0px; right: 0px; bottom: 0px; left: 0px;"
    >
      <img
        alt="Landscape photograph by Tobias Tullius"
        class="octo-UI-c-kxYyuZ"
        src="https://images.unsplash.com/photo-1535025183041-0991a977e25b?w=300&dpr=2&q=80"
      />
    </div>
  </div>
</div>
`;
